src/rgw/rgw_redis_queue.cc

```diff
@@ -0,0 +1,113 @@
+#include "rgw_redis_queue.h"
+
+namespace rgw {
+namespace redisqueue {
+
+int initQueue(boost::asio::io_context& io, connection* conn, config* cfg,
+              optional_yield y) {
+  return rgw::redis::loadLuaFunctions(io, conn, cfg, y);
+}
+
+// FIXME: Perhaps return the queue length in calls to reserve, commit, abort
+// etc and do not use this function explicitly?
+int queueStatus(connection* conn, const std::string& name,
+                std::tuple<int, int>& res, optional_yield y) {
+  boost::redis::request req;
+  boost::redis::response<int, int> resp;
+  boost::system::error_code ec;
+
+  try {
+    req.push("LLEN", "reserve:" + name);
+    req.push("LLEN", "queue:" + name);
+
+    rgw::redis::redis_exec(conn, ec, req, resp, y);
+    if (ec) {
+      std::cerr << "RGW Redis Queue:: " << __func__
+                << "(): ERROR: " << ec.message() << std::endl;
+      return -ec.value();
+    }
+    res = std::make_tuple(std::get<0>(resp).value(), std::get<1>(resp).value());
+    return 0;
+
+  } catch (const std::exception& e) {
+    std::cerr << "RGW Redis Queue:: " << __func__
+              << "(): Exception: " << e.what() << std::endl;
+    return -EINVAL;
+  }
+}
+
+int reserve(connection* conn, const std::string name, optional_yield y) {
+  boost::redis::request req;
+  rgw::redis::RedisResponseMap resp;
+
+  int reserveSize = 120;
```

Comment:
what does "120" mean?

Reply:
It's the number of characters to use in the string. 
Since I generate the string from the charset, they all take a single byte, allowing us to reserve approximately `reserveSize` bytes. 
https://redis.io/docs/latest/commands/memory-usage/

Reply:
but currently, the default reservations is, IIRC, 2KB

Reply:
I will make this reservation 2000 char long as well. 120 was randomly chosen.

Reply:
pleas check what size we currently use in the rgw_notify.cc